### PR TITLE
[ROCm][CI] Fix plugins test group; updating terratorch and dependencies

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
Follow up to: #34416 
This PR stabilizes the TerraTorch test dependencies and fixes a minor runtime warning.

**Dependency changes:**

- Replace the git commit pin for `terratorch` (`@07184fc...`) with the stable release `terratorch==1.2.2`, matching `test.txt` requirement versions.
- Pin `segmentation-models-pytorch==0.5.0` to ensure compatibility with `terratorch==1.2.2`. Previous unpinned versions of smp renamed the `use_batchnorm` parameter to `use_norm` in `UnetDecoder`, causing a `TypeError` at model initialization.
- Pin `imagehash==4.3.2`, required by the Prithvi plugin tests.

**Code minor patch:**

- In `vllm/model_executor/models/terratorch.py`, replace `torch.tensor(v)` with `torch.as_tensor(v).clone()` when constructing the `BatchFeature` in `TerratorchMultiModalProcessor.apply()`. This avoids the `UserWarning` triggered when copying from an existing tensor.

**Testing:**

- `plugins_tests/test_platform_plugins.py`
- `plugins_tests/test_io_processor_plugins.py`
- `plugins_tests/test_stats_logger_plugins.py`
- `plugins_tests/test_scheduler_plugins.py`
- `distributed/test_distributed_oot.py`
- `entrypoints/openai/test_oot_registration.py`
- `models/test_oot_registration.py`
- `plugins/lora_resolvers`

### CODE REVIEW

File: vllm/model_executor/models/terratorch.py [hunk 1]
```diff
@@ -208,7 +208,10 @@ def apply(
 
         _, passthrough_data = self._get_hf_mm_data(mm_items)
         mm_processed_data = BatchFeature(
-            {k: torch.tensor(v).unsqueeze(0) for k, v in passthrough_data.items()},
+            {
+                k: torch.as_tensor(v).clone().unsqueeze(0)
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply vllm/model_executor/models/terratorch.py]: Done :)

**Review by tjtanaa — APPROVED** [review]
LGTM

**Comment by DarkLight1337** [comment]
> This avoids the `UserWarning` triggered when copying from an existing tensor.

Maybe we can replace this with `(v if isinstance(v, torch.Tensor) else torch.tensor(v)).unsqueeze(0)`

**Comment by AndreasKaratzas** [comment]
> > This avoids the `UserWarning` triggered when copying from an existing tensor.
> 
> Maybe we can replace this with `(v if isinstance(v, torch.Tensor) else torch.tensor(v)).unsqueeze(0)`

Isn't `as_tensor` supposed to have the same conditional underneath?

**Comment by AndreasKaratzas** [comment]
> Hmm, can we get rid of the `clone` call then?

I think that  `.detach().clone()` is the idiomatic PyTorch pattern and exactly matches the original `torch.tensor()` semantics (which always detaches and copies). It costs nothing at runtime on a tensor that doesn't require grad.

**Comment by DarkLight1337** [comment]
What I mean here is that actually we can skip the copy

**Review by DarkLight1337 — APPROVED** [review]
